consensus/safety-rules/src/persistent_safety_storage.rs

```diff
@@ -16,32 +18,45 @@ use std::str::FromStr;
 /// @TODO add retrieval of private key based upon public key to persistent store
 pub struct PersistentSafetyStorage {
     internal_store: BoxedStorage,
+    execution_public_key: Option<Ed25519PublicKey>,
 }
 
 impl PersistentSafetyStorage {
-    pub fn in_memory(private_key: Ed25519PrivateKey) -> Self {
+    pub fn in_memory(
+        consensus_private_key: Ed25519PrivateKey,
+        execution_public_key: Option<Ed25519PublicKey>,
+    ) -> Self {
         let storage = BoxedStorage::from(InMemoryStorage::new());
-        Self::initialize(storage, private_key, Waypoint::default())
+        Self::initialize(
+            storage,
+            consensus_private_key,
+            execution_public_key,
+            Waypoint::default(),
+        )
     }
 
     /// Use this to instantiate a PersistentStorage for a new data store, one that has no
     /// SafetyRules values set.
     pub fn initialize(
         mut internal_store: BoxedStorage,
-        private_key: Ed25519PrivateKey,
+        consensus_private_key: Ed25519PrivateKey,
+        execution_public_key: Option<Ed25519PublicKey>,
         waypoint: Waypoint,
     ) -> Self {
-        Self::initialize_(&mut internal_store, private_key, waypoint)
+        Self::initialize_(&mut internal_store, consensus_private_key, waypoint)
             .expect("Unable to initialize backend storage");
-        Self { internal_store }
+        Self {
+            internal_store,
+            execution_public_key,
```

Comment:
if we go this path, let's just pass in the execution private key?

Reply:
some technical problems. I cannot fetch the private_key here I think.

Reply:
why the execution_public_key is not stored in the internal_store?